cpp/mg_utility/mg_graph.hpp

```diff
@@ -90,6 +90,28 @@ class Graph : public GraphView<TSize> {
     return neighbours_[node_id];
   }
 
+  /// Gets out-neighbour nodes.
+  ///
+  /// @param[in] node_id target node id
+  ///
+  /// @return vector of neighbours
+  const std::vector<TNeighbour> &OutNeighbours(TSize node_id) const {
+    return Neighbours(node_id);
+  }
+
+  /// Gets in-neighbour nodes.
+  ///
+  /// @param[in] node_id target node id
+  ///
+  /// @return vector of neighbours
+  const std::vector<TNeighbour> &InNeighbours(TSize node_id) const {
+    if (node_id < 0 && node_id >= nodes_.size()) {
+      throw mg_exception::InvalidIDException();
+    }
+
+    return in_neighbours_[node_id];
```

Comment:
Why the above method doesn't have the same condition? Maybe just check if there is node_id in `in_neighbours_`?

Reply:
Were you asking about `OutNeighbours()` not having the same condition? If so, that function solely calls `Neighbours()`, which does have the condition you’ve mentioned.

The check is handled by the `node_id >= nodes_.size()` condition since `in_neighbours_` contains data for each node in ascending order by `node_id`. This design is in line with how the contributors before me implemented `neighbours_`.